From 47247654066441b5dbd9096018a5d45bac584a6f Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Tue, 3 Jul 2018 15:28:29 -0700
Subject: [PATCH 0021/1051] mmc: cavium_thunderx: Use proper register to clear
 interrupts

The CSR at offset 0x80 (MIO_EMM_INT_W1S) actually sets interrupts instead
of clearing the interrupts. So use MIO_EMM_INT CSR which clears the
interrupts.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mmc/host/cavium-thunderx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/cavium-thunderx.c b/drivers/mmc/host/cavium-thunderx.c
index eee08d81b242..e6bbb76abe25 100644
--- a/drivers/mmc/host/cavium-thunderx.c
+++ b/drivers/mmc/host/cavium-thunderx.c
@@ -115,7 +115,7 @@ static int thunder_mmc_probe(struct pci_dev *pdev,
 	 * Clear out any pending interrupts that may be left over from
 	 * bootloader. Writing 1 to the bits clears them.
 	 */
-	writeq(127, host->base + MIO_EMM_INT_EN(host));
+	writeq(127, host->base + MIO_EMM_INT(host));
 	writeq(3, host->base + MIO_EMM_DMA_INT_ENA_W1C(host));
 	/* Clear DMA FIFO */
 	writeq(BIT_ULL(16), host->base + MIO_EMM_DMA_FIFO_CFG(host));
-- 
2.17.1

